### PR TITLE
Update to README.md

### PR DESCRIPTION
**Pull request by intelkevinputnam**
Proposed changes to improve readme.

1. Moved header to the top of the doc (above table of contents) along with some descriptive text.
2. Removed the "About" section as it didn't seem necessary after moving content to the top.
3. Editorial changes to existing content through line 130 or so.
4. Made some cosmetic changes to bullet list.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+Intel PMEM-CSI is a storage driver for container orchestrators like
+Kubernetes. It makes local persistent memory
+([PMEM](https://pmem.io/)) available as a filesystem volume to
+container applications. It can currently utilize non-volatile memory
+devices that can be controlled via the [libndctl utility
+library](https://github.com/pmem/ndctl). In this readme, we use
+*persistent memory* to refer to a non-volatile dual in-line memory
+module (NVDIMM).
+
+The [v0.6.0release](https://github.com/intel/pmem-csi/releases/tag/v0.6.0)
```
Comment (pohly) [inline README.md]: A space between "v0.6.0" and "release" got lost here.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+is the latest feature release and is [regularly updated](./DEVELOPMENT.md#release-management) with newer base images
+and bug fixes. Older versions are no longer supported.
+
+The PMEM-CSI driver follows the [CSI
+specification](https://github.com/container-storage-interface/spec) by
+listening for API requests and provisioning volumes accordingly.
+
+<h3>Table of Contents</h3>
```
Comment (pohly) [inline README.md]: Can we avoid HTML tags like `<h3>` in favor of normal markdown? Both works of course, markdown would just be more consistent because that's what we use for the other sections.

File: README.md [hunk 3]
```diff
@@ -68,69 +68,73 @@ should be used. Older releases are no longer supported.
...
-<sup>1 </sup> *Fragmented free space* state may develop via series of creation and deletion operations where the driver is no longer able to allocate a new namespace contiguously, although the free capacity (i.e. sum of available free section sizes) indicates the opposite. The PMEM-CSI driver is not capable of de-fragmenting the space or create allocation using combined smaller blocks. 
-A simplified example: This state develops after steps: create 63 GB namespace, create 1 GB namespace, delete 63 GB namespace. The free capacity is 127 GB, but the driver fails to create a namespace bigger than 64 GB. 
+<sup>1 </sup> **Free space fragmentation** is a problem when there appears to be
+enough free capacity for a new namespace, but there isn't a contiguous region
+big enough to allocate it. The PMEM-CSI driver is only capable of allocating
+continguous memory to a namespace and cannot de-fragment or combine smaller
+blocks. For example, this could happen when you create a 63 GB namespace,
+follwed by a 1 GB namespace, and then delete the 63 GB namespace. Eventhough
+there is 127 GB available, the driver cannot create a namespace larger than
+64 GB.
```
Comment (pohly) [inline README.md]: This can be enhanced further: the "there is 127 GB available" is stated without explaining from what that follows, i.e. the total capacity is missing. What about "this could happen when you create a 63 GB namespace in a region of 128 GB, followed by a 1 GB namespace, ..."?

"follwed" must be "followed". Funny how many different people must have read that already without noticing :joy:

File: README.md [hunk 4]
```diff
@@ -68,69 +68,73 @@ should be used. Older releases are no longer supported.
...
-when it [binds the device to a loop
-device](https://github.com/kubernetes/kubernetes/blob/7c87b5fb55ca096c007c8739d4657a5a4e29fb09/pkg/volume/util/util.go#L531-L534),
+details. `devdax` mode is not supported. Though a
+raw block volume would be useful when a filesystem isn't needed, Kubernetes cannot handle [binding a character device to a loop device](https://github.com/kubernetes/kubernetes/blob/7c87b5fb55ca096c007c8739d4657a5a4e29fb09/pkg/volume/util/util.go#L531-L534).
 
 ### LVM device mode
 
-The following diagram illustrates the operation in LVM device mode:
-![devicemode-lvm diagram](/docs/images/devicemodes/pmem-csi-lvm.png)
+In <abbr title="Logical Volume Management">LVM</abbr> mode the PMEM-CSI driver
```
Comment (pohly) [inline README.md]: What does `<abbr>` do? It doesn't seem to have an effect in the rendered document on GitHub.

File: README.md (hunk 3 above)
Comment (okartau) [reply README.md]: > Funny how many different people must have read that

High chance you are the first one reading this (new text added in this PR) in detail. And even for published version, I dare not hope for having many readers.

Comment (pohly) [reply README.md]: > High chance you are the first one reading this (new text added in this PR)

Oh, right. I thought it had been already in the branch.

> And even for published version, I dare not hope for having many readers.

What? You don't want people to use PMEM-CSI?! :scream:

Comment (okartau) [reply README.md]: let me rephrase, I hope,want for many but I estimate number is low despite our hopes.

File: README.md (hunk 4 above)
Comment (intelkevinputnam) [reply README.md]: It makes the meaning of abbreviation or acronym appear when hovering over it (like a tooltip). I'm surprised GitHub appears to be stripping it out. I'll remove it.

File: README.md (hunk 2 above)
Comment (intelkevinputnam) [reply README.md]: Yep.

File: README.md (hunk 1 above)
Comment (intelkevinputnam) [reply README.md]: Thx

File: README.md (hunk 3 above)
Comment (intelkevinputnam) [reply README.md]: Good catch. Thanks.
